### ISSUE TITLE
[Bug] Tesseract 4 download failed

### ISSUE THREAD

**Issue by OmrSi** [SubtitleEdit/subtitleedit#3172]
I was just now trying to download Tesseract 4 and this happened:
![screenshot 641](https://user-images.githubusercontent.com/20923700/48318124-7317d700-e604-11e8-9047-1f34a4e6f6d3.png)

I don't know If it's a problem from my end, I tried using a proxy and a clean settings file.

**Comment by OmrSi** [comment]
@niksedk  This was fixed?

**Comment by OmrSi** [comment]
I guess not, just tested the latest beta.

**Comment by niksedk** [comment]
Works fine here... what's your OS? Do you know what version of .net framework you have installed?

**Comment by OmrSi** [comment]
I'm running win 10, Redstone 3.
I was able to download it when you first retuned to 3, but now I'm trying to download it again and getting this, I tried a proxy and a clean settings file and a clean beta zip. 

I was able, however, yo download the files from here and add the folder to my app data, and it worked fine.

**Comment by OmrSi** [comment]
---------------------------

---------------------------
Download failed!



The remote name could not be resolved: 'raw.githubusercontent.com':    at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)

   at System.Net.WebClient.GetWebResponse(WebRequest request, IAsyncResult result)

   at System.Net.WebClient.DownloadBitsResponseCallback(IAsyncResult result)
---------------------------
OK   
---------------------------

**Comment by OmrSi** [comment]
@niksedk  I guess there is no use of this now?

**Comment by niksedk** [comment]
This will still be used... can you `ping raw.githubusercontent.com` from command line?

**Comment by OmrSi** [comment]
P.S: Downloading dictionaries works fine.
![screenshot 672](https://user-images.githubusercontent.com/20923700/48644610-cebed780-e9eb-11e8-8890-5bff2b9f8895.png)

**Comment by niksedk** [comment]
Does installing latest .net framework help ?
https://www.microsoft.com/net/download/dotnet-framework-runtime

**Comment by niksedk** [comment]
It should point to the download button on this url: https://github.com/SubtitleEdit/support-files/blob/master/Tesseract400.tar.gz
Code here: https://github.com/SubtitleEdit/subtitleedit/blob/master/src/Forms/Ocr/DownloadTesseract4.cs#L16

So it's a bit strange that it complains about `raw.githubusercontent.com`

**Comment by niksedk** [comment]
Yes, sounds weird.. can you download the tesseract dictionaries?

**Comment by OmrSi** [comment]
Well, something changes (or I was missing it), just to make sure of everything before I got back to you, I downloaded the latest beta and tried it alone, the download started, then I tried it with the installed version, clean settings file, It showed me the same error, but the dictionaries are downloading in both cases.

So, a stand-alone beta is working fine, but when I replace the installed with the beta, 4.0 doesn't download and shows the message also, it doesn't work If I opened the exe from the zip.

In other words, if I replace the installed exe it doesn't download, If I open it from the zip, It doesn't,
but if I extracted it to another location, it does.

let it noted that the download used to work from the zip and replaced the installed version and the dictionary download is working in all cases. 

Sorry If I wasted your time with this and it turned out to be a problem from my end or because of something I did. 😓😓

**Comment by OmrSi** [comment]
I uninstalled and reinstalled, the same thing, If I run it from the zip or from the replaced installed exe, the problem occurs, darnn tested and it worked fine with him.
So, I'll just download it from the extracted portable version, you can close this if you want...

**Comment by OmrSi** [comment]
I don't know if this changes anything:
![screenshot 684](https://user-images.githubusercontent.com/20923700/48665713-19a82000-eabc-11e8-9240-58b70463e08e.png)

**Comment by niksedk** [comment]
@OmrSi: Last screenshot is from XP or .net framework 4.0?

**Comment by OmrSi** [comment]
No, My windows 10. :\
![screenshot 711](https://user-images.githubusercontent.com/20923700/48724217-492c6900-ec31-11e8-9a31-55f6f62e3e47.png)

This is, it's only showing 4.0:
![screenshot 712](https://user-images.githubusercontent.com/20923700/48724237-5fd2c000-ec31-11e8-90cc-0b86e37acbb7.png)

But when I try installing any higher version:
![screenshot 713](https://user-images.githubusercontent.com/20923700/48724339-a58f8880-ec31-11e8-8c10-3e6ed642b442.png)

**Comment by OmrSi** [comment]
@niksedk If you'd like, I can give you access to my computer using Anydesk or any other program you'd like to test it yourself. Just an idea...

**Comment by OmrSi** [comment]
I'll just close this since no one else is facing it.

**Comment by OmrSi** [comment]
@AggelosM The problem is still here, too.
And what you said isn't a fix per se.
I did the same when I needed it, but also, if you open a portable version and try, the download will work with no problems.
This issue is only happening when I try downloading from the installed version.
But since no one else is facing it, I guess we'll have to stick to these workarounds.
Also, the download is a one-time-only thing, so it's not a big issue...

**Comment by niksedk** [comment]
@OmrSi: Could you try latest beta - and download Tesseract 4.0.1:
https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.10/SubtitleEditBeta.zip

If this does not work, I can try building with .net framework 4.5.2...

**Comment by OmrSi** [comment]
Nope. :\
![{01CD2CEE-616A-4074-80A9-762530034D8A} png](https://user-images.githubusercontent.com/20923700/63537557-b18cdb80-c51e-11e9-89ca-c0282ace87b4.jpg)

**Comment by niksedk** [comment]
thx for testing though
What about this: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.10/SubtitleEditBeta-452.zip

**Comment by OmrSi** [comment]
Hmm, so this is gonna sound weird, but here is what happened.
I tried this and it didn't work, then I thought I'd try to run it as administrator, I did that and then I opened a file from inside SE and it downloaded the file just fine.

I went back to 3.5.10, and then I figured it out.
When you open a file from the Open option inside SE, the download works fine.
When you open a file from outside SE, it doesn't. ^^"
Hope that helps.

**Comment by xylographe** [comment]
Probably unrelated, but when I had trouble with SSL/TLS connection in a script, I solved it with,
```
[System.Net.ServicePointManager]::SecurityProtocol = 'Tls12, Ssl3'
[System.Net.ServicePointManager]::Expect100Continue = $true
```
before starting the download.

**Comment by niksedk** [comment]
@OmrSi: nice catch :)

It's surely connected to `SecurityProtocol`... which was not called when opening an image based sub.
 
How is this: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.10/SubtitleEditBeta.zip

**Comment by OmrSi** [comment]
Works great now. :)
Thank you. ^_^

**Comment by 1024mb** [comment]
I was having the same problem since I downloaded and installed latest stable version. With the beta it's finally working. Thanks.
Also I want to ask if there is any way we can use tesseract alpha 5 with SE? I mean, surely we can't for API reasons but can't hurt to ask.
